File: src/main/java/org/fountainmc/api/event/world/ChunkLoadEvent.java
package org.fountainmc.api.event.world;

import javax.annotation.Nonnull;

import org.fountainmc.api.world.Chunk;

import static com.google.common.base.Preconditions.checkNotNull;

public interface ChunkLoadEvent extends ChunkEvent {

    @Nonnull
    static ChunkLoadEvent create(@Nonnull Chunk chunk) {
        checkNotNull(chunk, "Null chunk");
        return () -> chunk;
    }

}

File: src/main/kotlin/org/fountainmc/api/event/world/ChunkLoadEvent.kt
package org.fountainmc.api.event.world

import org.fountainmc.api.world.Chunk

interface ChunkLoadEvent : ChunkEvent {
    companion object {
        fun create(chunk: Chunk): ChunkLoadEvent {
            class SimpleChunkLoadEvent(override val chunk: Chunk): ChunkLoadEvent
            return SimpleChunkLoadEvent(chunk)
        }
    }
}
